src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -405,18 +405,18 @@ private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol in
                         if ((InConstructorInitializer || InFieldInitializer) && ContainingMemberOrLambda.ContainingSymbol.Kind == SymbolKind.NamedType)
                             CheckFeatureAvailability(designation, MessageID.IDS_FeatureExpressionVariablesInQueriesAndInitializers, diagnostics);
 
-                        localSymbol.SetTypeWithAnnotations(TypeWithAnnotations.Create(declType));
-                        localSymbol.SetValEscape(GetValEscape(declType, inputValEscape));
+                        localSymbol.SetTypeWithAnnotations(declType);
+                        localSymbol.SetValEscape(GetValEscape(declType.Type, inputValEscape));
 
                         // Check for variable declaration errors.
                         hasErrors |= localSymbol.ScopeBinder.ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
 
                         if (!hasErrors)
-                            hasErrors = CheckRestrictedTypeInAsync(this.ContainingMemberOrLambda, declType, diagnostics, typeSyntax ?? (SyntaxNode)designation);
+                            hasErrors = CheckRestrictedTypeInAsync(this.ContainingMemberOrLambda, declType.Type, diagnostics, typeSyntax ?? (SyntaxNode)designation);
 
                         variableSymbol = localSymbol;
                         variableAccess = new BoundLocal(
```

Comment:
No, it is because bound nodes do not have `TypeWithAnnotations` as the type, only a `TypeSymbol` as the type.  The variable named `declType` changed from being a `TypeSymbol` to a `TypeWithAnnotations`.

---
In reply to: [269743097](https://github.com/dotnet/roslyn/pull/34273#discussion_r269743097) [](ancestors = 269743097)